data/scripts.js

```diff
@@ -1353,7 +1358,7 @@ exports.BattleScripts = {
 				}
 
 				// Remove rejected moves from the move list.
-				if (rejected && movePool.length) {
+				if (rejected && movePool.length - availableHP + ((!availableHP || hasMove['hiddenpower'] && move.id !== 'hiddenpower') ? 0 : 1) >= 5 - moves.length) {
```

Comment:
Except that's not quite the same...
I could make it multiline, but I want to make it clear that this is a single mathematical-logical unit.